wasm/jit/engine.go

```diff
@@ -446,3 +447,230 @@ func (e *engine) builtinFunctionMemoryGrow(f *compiledWasmFunction) {
+	ir, err := wazeroir.Compile(f)
+	if err != nil {
+		return nil, fmt.Errorf("failed to lower to wazeroir: %w", err)
+	}
+
+	compiler, err := newCompiler(e, f, ir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize assembly builder: %w", err)
+	}
+
+	compiler.emitPreamble()
+
+	for _, op := range ir.Operations {
+		switch o := op.(type) {
+		case *wazeroir.OperationUnreachable:
+			compiler.compileUnreachable()
+		case *wazeroir.OperationLabel:
+			if err := compiler.compileLabel(o); err != nil {
+				return nil, fmt.Errorf("error handling label operation: %w", err)
+			}
+		case *wazeroir.OperationBr:
+			if err := compiler.compileBr(o); err != nil {
+				return nil, fmt.Errorf("error handling br operation: %w", err)
+			}
+		case *wazeroir.OperationBrIf:
+			if err := compiler.compileBrIf(o); err != nil {
+				return nil, fmt.Errorf("error handling br_if operation: %w", err)
+			}
+		case *wazeroir.OperationBrTable:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationCall:
+			if err := compiler.compileCall(o); err != nil {
+				return nil, fmt.Errorf("error handling call operation: %w", err)
+			}
+		case *wazeroir.OperationCallIndirect:
+			return nil, fmt.Errorf("unsupported operation in JIT compiler: %v", o)
+		case *wazeroir.OperationDrop:
+			if err := compiler.compileDrop(o); err != nil {
+				return nil, fmt.Errorf("error handling drop operation: %w", err)
+			}
+		case *wazeroir.OperationSelect:
+			if err := compiler.compileSelect(); err != nil {
+				return nil, fmt.Errorf("error handling select operation: %w", err)
+			}
+		case *wazeroir.OperationPick:
+			if err := compiler.compilePick(o); err != nil {
+				return nil, fmt.Errorf("error handling pick operation: %w", err)
+			}
+		case *wazeroir.OperationSwap:
+			if err := compiler.compileSwap(o); err != nil {
+				return nil, fmt.Errorf("error handling swap operation: %w", err)
+			}
+		case *wazeroir.OperationGlobalGet:
+			if err := compiler.compileGlobalGet(o); err != nil {
+				return nil, fmt.Errorf("error handling global.get operation: %w", err)
+			}
+		case *wazeroir.OperationGlobalSet:
+			if err := compiler.compileGlobalSet(o); err != nil {
+				return nil, fmt.Errorf("error handling global.set operation: %w", err)
+			}
+		case *wazeroir.OperationLoad:
+			if err := compiler.compileLoad(o); err != nil {
+				return nil, fmt.Errorf("error handling load operation: %w", err)
+			}
+		case *wazeroir.OperationLoad8:
+			if err := compiler.compileLoad8(o); err != nil {
+				return nil, fmt.Errorf("error handling load8 operation: %w", err)
+			}
+		case *wazeroir.OperationLoad16:
+			if err := compiler.compileLoad16(o); err != nil {
+				return nil, fmt.Errorf("error handling load16 operation: %w", err)
+			}
+		case *wazeroir.OperationLoad32:
+			if err := compiler.compileLoad32(o); err != nil {
+				return nil, fmt.Errorf("error handling load16 operation: %w", err)
```

Comment:
I think the formatting can be templated

```suggestion
				return nil, fmt.Errorf("error handling load32 operation: %w", err)
```